tests/test_edgeql_enums.py

```diff
@@ -188,3 +188,27 @@ async def test_edgeql_enums_assignment_04(self):
                 'color': 'GREEN',
             }],
         )
+
+    async def test_edgeql_enums_json_cast_01(self):
+        self.assertEqual(
+            await self.con.query(
+                "SELECT <json><test::color_enum_t>'RED'"
+            ),
+            ['"RED"'])
+
+        await self.assert_query_result(
+            "SELECT <test::color_enum_t><json>'RED'",
+            ['RED'])
```

Comment:
`self.assert_query_result` checks both JSON and binary compile modes automatically. SO I'd also add a test without an explicit `<json>` cast, testing that JSON mode handles enums just fine too:

```
        await self.assert_query_result(
            "SELECT <test::color_enum_t>'RED'",
            ['RED'])
```

Reply:
I can add that test, but it looks like it already passes even without my changes